server/api/src/rsGenQuery.cpp

```diff
@@ -805,54 +794,50 @@ _rsGenQuery( rsComm_t *rsComm, genQueryInp_t *genQueryInp,
     *genQueryOut = ( genQueryOut_t* )malloc( sizeof( genQueryOut_t ) );
     memset( ( char * )*genQueryOut, 0, sizeof( genQueryOut_t ) );
 
-    if ( ruleExecuted == 0 ) {
-        memset( ( char* )&rei, 0, sizeof( rei ) );
-        /* Include the user info for possible use by the rule.  Note
-        that when this is called (as the agent is initializing),
-        this user info is not confirmed yet.  For password
-        authentication though, the agent will soon exit if this
-        is not valid.  But for GSI, the user information may not
-        be present and/or may be changed when the authentication
-        completes, so it may not be safe to use this in a GSI
-        enabled environment.  This addition of user information
-        was requested by ARCS/IVEC (Sean Fleming) to avoid a
-        local patch.
-        */
-        rei.uoic = &rsComm->clientUser;
-        rei.uoip = &rsComm->proxyUser;
-
-        status = applyRule( "acAclPolicy", NULL, &rei, NO_SAVE_REI );
-        if ( status == 0 ) {
-
-            ruleExecuted = 1; /* No need to retry next time since it
-                             succeeded.  Since this is called at
-                             startup, the Rule Engine may not be
-                             initialized yet, in which case the
-                             default setting is fine and we should
-                             retry next time. */
+    if (ruleExecuted == 0) {
+        std::string svc_role;
+        irods::error ret = get_catalog_service_role(svc_role);
+        int status{};
+        if (!ret.ok()) {
+            irods::log(PASS(ret));
+            // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
+            status = ret.code();
         }
-    }
 
-    // =-=-=-=-=-=-=-
-    // verify that we are running a query for another agent connection
-    const bool agent_conn_flag = irods::server_property_exists(irods::AGENT_CONN_KW);
+        if (irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role) {
+            constexpr auto priv{0};
+            constexpr auto control_flag{2};
 
-    // =-=-=-=-=-=-=-
-    // detect if a request for disable of strict acls is made
-    int acl_val = -1;
-    char* dis_kw = getValByKey( &genQueryInp->condInput, DISABLE_STRICT_ACL_KW );
-    if ( agent_conn_flag && dis_kw ) {
-        acl_val = 0;
+            // Call directly to avoid additional policy
```

Comment:
This call is still going to trigger at least one database PEP (`pep_database_gen_query_access_control_setup_*`), so I'm not sure that I understand this comment. Please explain.

Reply:
That comment was associated with this change: https://github.com/irods/irods/pull/8264#discussion_r1994092194

It was reverted since the code wouldn't compile. Will remove if we continue to use `chlGenQueryAccessControlSetup(...)`.

Reply:
Removed.